jsk_perception/src/remove_blurred_frames.cpp

```diff
@@ -0,0 +1,97 @@
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/o2r other materials provided
+ *     with the distribution.
+ *   * Neither the name of the JSK Lab nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+/*
+ * remove_blurred_frames.cpp
+ * Author: Yoshiki Obinata <obinata@jsk.imi.i.u-tokyo.ac.jp>
+ */
+
+#include "jsk_perception/remove_blurred_frames.h"
+
+namespace jsk_perception{
+    void RemoveBlurredFrames::onInit(){
+        DiagnosticNodelet::onInit();
+        srv_ = boost::make_shared<dynamic_reconfigure::Server<Config> > (*pnh_);
+        dynamic_reconfigure::Server<Config>::CallbackType f =
+            boost::bind(&RemoveBlurredFrames::configCallback, this, _1, _2);
+        srv_->setCallback (f);
+        pub_ = advertise<sensor_msgs::Image>(*pnh_, "output", 1);
+        pub_masked_ = advertise<sensor_msgs::Image>(*pnh_, "output/mask", 1);
+        pub_var_ = advertise<std_msgs::Float64>(*pnh_, "output/var", 1);
+        onInitPostProcess();
+    }
+
+    void RemoveBlurredFrames::subscribe(){
+        sub_ = pnh_->subscribe("input", 1, &RemoveBlurredFrames::work, this);
+    }
+
+    void RemoveBlurredFrames::unsubscribe(){
+        sub_.shutdown();
+    }
+
+    void RemoveBlurredFrames::work(const sensor_msgs::Image::ConstPtr& image_msg){
+        cv::Mat image, gray, laplacian_image;
+        sensor_msgs::Image::Ptr mask_img_msg;
+        cv::Scalar mean, stddev;
+        std_msgs::Float64 var_msg;
+        double var;
+        vital_checker_ -> poke();
+        boost::mutex::scoped_lock lock(mutex_);
+        image = cv_bridge::toCvShare(image_msg, image_msg->encoding) -> image;
```

Comment:
Could you specify rgb encoding? R and B seem to be incompatible.
> RGB[A] to Gray:Y←0.299⋅R+0.587⋅G+0.114⋅B

https://docs.opencv.org/4.1.0/de/d25/imgproc_color_conversions.html
https://github.com/opencv/opencv/blob/347246901eccabe503985a64f16813ca859af25a/modules/imgproc/src/color.cpp#L211-L214

Reply:
Do you mean as both `COLOR_BGR2GRAY` and `COLOR_RGB2GRAY` use same code, we shouldn't use `cv::cvtColor`

Reply:
And, is it the bug of opencv?

Reply:
https://github.com/opencv/opencv/blob/17234f82d025e3bbfbf611089637e5aa2038e7b8/modules/imgproc/src/color.hpp#L54 ??

Reply:
To convert RGB to GRAY in the next line, the encoding have to be RGB, not as input is.

Reply:
divide the next line like RGB2GRAY and BGR2GRAY, or specify RGB

Reply:
I see, the line `cv::cvtColor(image, gray, cv::COLOR_RGB2GRAY);` . Sorry for my overlooking.